sympy/series/limits.py

```diff
@@ -329,6 +327,31 @@ def remove_abs(expr):
                 r = gruntz(e, z, z0, dir)
             if r is S.NaN or l is S.NaN:
                 raise PoleError()
+        except SignException as er:
+            sig = er.sign
+            r = None
+            # seq is True if and only if the function call originates from limitseq.py
+            # rather than limits.py. In the former case, we return r as normal.
+            if (hints.get('seq', True)):
+                return r
+            # The purpose of exception handling here is to attempt to compute limits of expressions
+            # which do not exist for non-integer values (i.e. expressions like (-1)**x). The solution
+            # is to compute the limit using limit_seq() from the limitseq module, but this is only
+            # valid if we are computing a limit to infinity and, currently, if the expression has no
+            # complex terms.
+            if (z0 is not S.Infinity):
+                raise NotImplementedError('Result depending on the sign of %s must be a limit toward oo' % sig)
+                return r
+            from sympy.core.numbers import I
+            if (e.has(I) or len(e.free_symbols) > 1):
```

Comment:
It's not necessary to put parentheses around the expression after an `if` or `except` (below).